Automattic/wp-calypso
# Theme Showcase: New Tiers Filtering

Tracking issue: https://github.com/Automattic/dotcom-forge/issues/4597

## Proposed Changes

* Introduce the new tier system to the Theme Showcase routing and pricing filtering.

<img width="873" alt="Screenshot 2023-12-13 at 16 58 24" src="https://github.com/Automattic/wp-calypso/assets/2070010/cd10a36b-2901-48d1-bda2-c3016c6403fa">

## Testing Instructions



* Enable the `themes/tiers` feature flag.
* Patch D131846-code and sandbox the API.
* Navigate to the Theme Showcase.
* Ensure the pricing tier dropdown shows all the new tiers (except Sensei, for now — this is intended).
* Try choosing Personal.
* Ensure the list only shows the Screenplay theme.
* Try opening URLs for all the tiers (e.g. `/themes/woocommerce/{ SITE }`.
* Ensure the Showcase opens correctly. Ignore that the list of themes doesn't correspond with the filter (that's for D131846-code).

## Pre-merge Checklist
